arraycontext/container/traversal.py

```diff
@@ -741,7 +741,10 @@ def _unflatten(template_subary: ArrayOrContainerT) -> ArrayOrContainerT:
             # {{{ check strides
 
             if strict and hasattr(template_subary, "strides"):
-                if template_subary.strides != subary.strides:
+                if (
+                    (template_subary.strides != subary.strides)
+                    and (template_subary.size != 0)
```

Comment:
Please add a comment explaining why checking strides on 0-size arrays is ill-defined.

Reply:
Added comment